tm2/pkg/bft/abci/types/types.go

```diff
@@ -206,6 +206,7 @@ type Error interface {
 
 type Event interface {
 	AssertABCIEvent()
+	SetMsgIdx(int) interface{}
```

Comment:
If you want to force that an Event needs an ID, I would recommend adding the method providing that ID instead of the provided Setter:

```suggestion
	ID() []byte
```

If the ID is expected to be sequential:

```suggestion
	Seq() uint64
```

Reply:
Looking Good!

Reply:
Reverted.
Event needs not only ID, but also setter for setting msg_idx.

Currently, event gets emitted from here
https://github.com/gnolang/gno/blob/d7c04bee3c90718bf8b82a00399f6aa387e82376/gnovm/stdlibs/std/emit_event.go#L14-L31
At this point, there is no way for us to determine message index if current request was made by multi-msg tx.

However `baseapp.go` does know full list of messages so, we can set message index from there.
https://github.com/gnolang/gno/blob/dabf13c705b4ac13ffd25d55a9a4a2107e335555/tm2/pkg/sdk/baseapp.go#L634-L666